### PR TITLE
Support configuring cgroupMode

### PR DESCRIPTION
**Pull request by DebakelOrakel**
## Checklist

- [x] The PR has a meaningful title. It will be used to auto generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [ ] Link this PR to related issues or PRs.

### CODE REVIEW

**Review by HappyTetrahedron — APPROVED** [review]
LGTM

File: component/node-config.jsonnet [hunk 1]
```diff
@@ -0,0 +1,15 @@
+local common = import 'common.libsonnet';
+local com = import 'lib/commodore.libjsonnet';
+local kap = import 'lib/kapitan.libjsonnet';
+local kube = import 'lib/kube.libjsonnet';
+local inv = kap.inventory();
+
+local params = inv.parameters.openshift4_nodes;
+
+local nodeConfig = kube._Object('config.openshift.io/v1', 'Node', 'cluster') {
```
Comment (simu) [inline component/node-config.jsonnet:9]: I'd adjust this to also add the component common labels (see e.g. https://github.com/appuio/component-openshift4-nodes/blob/928eb622fbad18a29d8c5d6e106e61097dad5a0c/component/container-runtime.jsonnet#L13-L17):

```suggestion
local nodeConfig = kube._Object('config.openshift.io/v1', 'Node', 'cluster') {
  metadata+: {
    labels+: common.DefaultLabels,
  },
```

File: docs/modules/ROOT/pages/references/parameters.adoc [hunk 2]
```diff
@@ -72,6 +71,21 @@ The namespace where machine-api related objects should be created.
...
+default::
++
+[source,yaml]
+----
+cgroupMode: v1
+----
+
+A dictionary holding the default values applied to the `node.config.openshift.io` object created by this component.
+
+You can switch between cgroup v1 and cgroup v2, as needed, by editing the node.config object. For more information, see https://docs.openshift.com/container-platform/latest/installing/install_config/enabling-cgroup-v2.html[Enabling Linux control group version 2].
```
Comment (simu) [inline docs/modules/ROOT/pages/references/parameters.adoc:87]:
```suggestion
You can switch between cgroup v1 and cgroup v2, as needed, by editing the node.config object.
For more information, see https://docs.openshift.com/container-platform/latest/installing/install_config/enabling-cgroup-v2.html[Enabling Linux control group version 2].
```

**Review by simu — APPROVED** [review]
LGTM overall, see inline comments for minor improvements

**Pull request merged**